File: core/src/test/java/com/oneandone/typedrest/ConsumerEndpointTest.java
package com.oneandone.typedrest;

import static com.github.tomakehurst.wiremock.client.WireMock.*;
import static org.apache.http.HttpStatus.*;
import org.junit.*;

public class ConsumerEndpointTest extends AbstractEndpointTest {

    private ConsumerEndpoint<MockEntity> endpoint;

    @Before
    @Override
    public void before() {
        super.before();
        endpoint = new ConsumerEndpointImpl<>(entryEndpoint, "endpoint", MockEntity.class);
    }

    @Test
    @Ignore("Works in isolation but fails when executed as part of test suite")
    public void testTrigger() throws Exception {
        stubFor(post(urlEqualTo("/endpoint"))
                .withRequestBody(equalToJson("{\"id\":1,\"name\":\"input\"}"))
                .willReturn(aResponse().withStatus(SC_OK)));
 
       endpoint.trigger(new MockEntity(1, "input"));
    }
}

File: typedrest-serializers-jackson/src/test/kotlin/net/typedrest/endpoints/rpc/ConsumerEndpointTest.kt
package net.typedrest.endpoints.rpc

import net.typedrest.endpoints.AbstractEndpointTest
import net.typedrest.MockEntity
import net.typedrest.http.HttpMethod
import net.typedrest.tests.*
import okhttp3.mockwebserver.MockResponse
import kotlin.test.*

class ConsumerEndpointTest : AbstractEndpointTest() {
    private val endpoint = ConsumerEndpointImpl(entryEndpoint, "endpoint", MockEntity::class.java)

    @Test
    fun testInvoke() {
        server.enqueue(MockResponse())

        endpoint.invoke(MockEntity(1, "input"))
        server.assertRequest(HttpMethod.POST)
            .withJsonBody("""{"id":1,"name":"input"}""")
    }
}
